src/Raven.Server/Documents/RemoteAttachmentsStorage.cs

```diff
@@ -374,4 +338,84 @@ public RemoteAttachmentsSender UpdateRemoteAttachmentsFromDatabaseRecord(Databas
             return null;
         }
     }
+
+    public int ProcessRemoteAttachments(DocumentsOperationContext context, DateTime currentTime, Dictionary<string, List<Slice>> seenDocs, Dictionary<string, Dictionary<string, AttachmentRemoteInfo>> attachmentsToUploadByIdentifier)
+    {
+        var processedCount = 0;
+        var docsTree = context.Transaction.InnerTransaction.ReadTree(_treeName);
+        foreach (var (docId, allTicks) in seenDocs)
+        {
+            using (DocumentIdWorker.GetLoweredIdSliceFromId(context, docId, out var lowerId))
+            using (var doc = Database.DocumentsStorage.Get(context, lowerId, DocumentFields.Data | DocumentFields.Id, throwOnConflict: true))
+            {
+                // remove all entries for processed document
+                foreach (var ticks in allTicks)
+                {
+                    docsTree.MultiDelete(ticks, lowerId);
+                }
+
+                if (doc == null || doc.TryGetMetadata(out var metadata) == false || metadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray attachments) == false)
+                {
+                    // doc was deleted
+                    continue;
+                }
+
+                // lets go over attachments and mark those we proceed as remote
+                foreach (BlittableJsonReaderObject attachmentInMetadata in attachments)
+                {
+                    if (attachmentInMetadata.TryGet(nameof(AttachmentName.RemoteParameters), out BlittableJsonReaderObject remoteParamsObject) == false)
+                        continue;
+
+                    if (remoteParamsObject.TryGet(nameof(RemoteAttachmentParameters.Flags), out object flagsObj) && Enum.TryParse(flagsObj.ToString(), out RemoteAttachmentFlags flags) && flags == RemoteAttachmentFlags.Remote)
+                    {
+                        // this attachment is already remote
+                        continue;
+                    }
```

Comment:
Do we still need to call `MarkAsRemote` here? Or check that?
What happens if we got `Flags = Remote` via replication? Did we delete the local attachment?

Reply:
I think this is no-op, because if we got `Flags = Remote` via replication, it means it went through `AttachmentsStorage.PutDirect()` and the local attachment was deleted there. 

In general we enter this if clause in two states 
1. old remoted attachment 
2. attachment that was uploaded by other node, while we was processing it

In both cases the local steam should be deleted & metadata updated. I think we can check if the item is already remote && attachmentsToUploadByIdentifier contains it, then we can add a log here that remote attachment was remoted by other node while we was processing.